slam-libraries/viam-cartographer/src/slam_service/slam_service_test.cc

```diff
@@ -69,45 +72,308 @@ BOOST_AUTO_TEST_CASE(OverwriteMapBuilderParameters_set_values_mapping) {
     slamService.occupied_space_weight = 9990.09;
     slamService.translation_weight = 9989.89;
     slamService.rotation_weight = 9988.88;
+
+    BOOST_TEST(slamService.GetActionMode() == SLAMServiceActionMode::MAPPING);
     slamService.OverwriteMapBuilderParameters();
     checkCartoMapBuilderParameters(slamService);
 }
 
 BOOST_AUTO_TEST_CASE(
     OverwriteMapBuilderParameters_check_default_values_mapping) {
+    // Mapping is the default action_mode when slamService is created
+    SLAMServiceImpl slamService;
+
+    BOOST_TEST(slamService.GetActionMode() == SLAMServiceActionMode::MAPPING);
+    slamService.OverwriteMapBuilderParameters();
+    checkCartoMapBuilderParameters(slamService);
+}
+
+BOOST_AUTO_TEST_CASE(OverwriteMapBuilderParameters_set_values_updating) {
+    SLAMServiceImpl slamService;
+    slamService.optimize_every_n_nodes = 9999;
```

Comment:
Yes, they're outlandish to ensure we don't end up with a flaky test. A flaky test could be created if we'd test with values here that would be realistic to set in the slam service. 

For example, we might be setting `max_submaps_to_keep = 4` here since we know that the default value is `max_submaps_to_keep = 3`. We then expect `OverwriteMapBuilderParameters` to successfully change the value from a `3` to a `4`. The test would lose it's power to make sure the change happens if we for some reason would decide to change the default value in the slam service from a `3` to a `4`.